apps/studio/components/grid/components/formatter/ForeignKeyFormatter.tsx

```diff
@@ -23,11 +23,13 @@ export const ForeignKeyFormatter = (props: Props) => {
   const { projectRef, tableId, row, column } = props
   const id = tableId ? Number(tableId) : undefined
 
-  const { data: selectedTable } = useTableQuery({
+  const { data } = useTableEditorQuery({
     projectRef: project?.ref,
     connectionString: project?.connectionString,
     id,
   })
+  const selectedTable = data?.table_info
```

Comment:
I'm just leaving annotations for myself to double check when I test your PR later so feel free to ignore (unless you have the answer at hand hahah)

do we not need `getTableLikeFromTableEditor` like here? need to check what's the diff between `getTableLikeFromTableEditor(data)` and `data.table_info`

Reply:
Yeah I'm actually not sure of the answer here either... I did it this way because before it was only fetching the table and not handling all the other types...